ipinfohigherlimit/widget.go

```diff
@@ -0,0 +1,135 @@
+package ipinfohigherlimit
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"text/template"
+
+	"bytes"
+
+	"github.com/olebedev/config"
+	"github.com/senorprogrammer/wtf/wtf"
+)
+
+// Config is a pointer to the global config object
+var Config *config.Config
+
+// Widget widget struct
+type Widget struct {
+	wtf.TextWidget
+	result string
+	colors struct {
+		name, value string
+	}
+}
+
+type ipinfo struct {
+	Query        string  `json:"query"`
+	ISP          string  `json:"isp"`
+	AS           string  `json:"as"`
+	City         string  `json:"city"`
+	Region       string  `json:"region"`
+	Country      string  `json:"country"`
+	CountryCode  string  `json:"countryCode"`
+	Latitude     float64 `json:"lat"`
+	Longitude    float64 `json:"lon"`
+	PostalCode   string  `json:"zip"`
+	Organization string  `json:"org"`
+	Timezone     string  `json:"timezone"`
+}
+
+// NewWidget constructor
+func NewWidget() *Widget {
+	widget := Widget{
+		TextWidget: wtf.NewTextWidget(" IPInfo ", "ipinfohigherlimit", false),
+	}
+
+	widget.View.SetWrap(false)
+
+	widget.config()
+
+	return &widget
+}
+
+// Refresh refresh the module
+func (widget *Widget) Refresh() {
+	widget.UpdateRefreshedAt()
+	widget.ipinfo()
+	widget.View.SetText(widget.result)
+}
+
+//this method reads the config and calls ipinfo for ip information
+func (widget *Widget) ipinfo() {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", "http://ip-api.com/json", nil)
+	if err != nil {
+		widget.result = fmt.Sprintf("%s", err.Error())
+		return
+	}
+	req.Header.Set("User-Agent", "curl")
+	response, err := client.Do(req)
+	if err != nil {
+		widget.result = fmt.Sprintf("%s", err.Error())
+		return
+	}
+	defer response.Body.Close()
+	if err != nil {
```

Comment:
```
if err != nil {
	widget.result = fmt.Sprintf("%s", err.Error())
	return
}
```
Sorry I can't see the reason for checking the same `err` two times(lines 73-76 & 78-81)

Reply:
Sorry, this part is from the original ipinfo. So I didn't change it. I will correct it.